full-service/migrations/2023-08-14-214222_authenticated_sender_memos/up.sql

```diff
@@ -0,0 +1,11 @@
+ALTER TABLE txos ADD COLUMN memo_type INTEGER NULL;
+
+CREATE TABLE authenticated_sender_memos (
+  txo_id TEXT PRIMARY KEY NOT NULL,
+  sender_address_hash TEXT NOT NULL,
+  payment_request_id UNSIGNED BIG INT,
+  payment_intent_id UNSIGNED BIG INT,
+  FOREIGN KEY (txo_id) REFERENCES txos(id)
+);
+
+UPDATE accounts SET next_block_index = 0;
```

Comment:
This seems out of place here. If it's a temp change for local dev, please mark it as such
```suggestion
// DO NOT MERGE: local testing
UPDATE accounts SET next_block_index = 0;

```

Reply:
How was this tested?
How do we know this doesn't break integrations that don't expect the next_block_index/account_block_height to ever decrease?

Reply:
I have tested it several times, it works as expected.

As for not breaking integrations, I don't feel like that's something to worry about. Do you think it's reasonable to assume that someone would be using this in a way that would cause their integration to break? Secondly, do we care if that is the case?

If we care, it's possible to do this another way but will require more work. I'm fine doing that but it doesn't seem worth the effort since the account will resync in a few minutes at current block height.

The other proposed solution was to run a query after the migration and iterate over all of the txos, manually calling the update, which would avoid requiring a resync.

Reply:
I have no idea how people are using full-service, but given that it is used by exchanges we very much care about breaking things for them.
This feels like a questionable approach to me, for example it appears that right now if you did it twice in a row (`UPDATE accounts SET next_block_index = 0;`, start full-service and let it do its thing, `UPDATE accounts SET next_block_index = 0;` again) you will misbehave since `add_authenticated_memo_to_database` will be called twice.

I am worried of any other potential incidents where something like this might get missed.

How difficult would it be to add code before [running migrations](https://github.com/mobilecoinofficial/full-service/blob/main/full-service/src/bin/main.rs#L85) that checks if this migration is about to run, and if so then after it runs goes and processes all known TxOuts in the database for their memos?

Its possible I am being over-conservative but my reason for this is that the cost of accidentally breaking one of the exchanges integrations or databases because something we didn't predict had happened would be very high...

Reply:
It isn't much work, maybe a few hours at most? I do agree that we want to avoid breaking things especially for exchanges, so I'll go ahead and implement this instead :) Thanks for being on top of these types of potential issues and bringing them up, appreciate it!

Reply:
Thank you!